backstory/src/test/java/com/google/sps/perspective/PerspectiveAPIClientTest.java

```diff
@@ -145,5 +151,34 @@ public void checkRightScores() {
     }
   }
 
-  // TODO: add a test that checks that the correct AnalyzeCommentRequest is passed to PerspectiveAPI
+  /**
+   * Check that the correct AnalyzeCommentRequest is passed to the Perspective API 
+   * by checking all values of AnalyzeCommentRequest that are set by PerspectiveAPIClient.
+   */
+  @Test
+  public void checkRequest() {
+    MockPerspectiveAPIFactory factory = new MockPerspectiveAPIFactory(new HashMap<AttributeType, Float>()); 
+    PerspectiveAPI mockAPI = factory.newInstance();
+    PerspectiveAPIClient client = new PerspectiveAPIClient(mockAPI);
+
+    PerspectiveValues values = client.analyze(DESIRED_TYPES, DEFAULT_TEXT);
+
+    ArgumentCaptor<AnalyzeCommentRequest> requestCaptor = ArgumentCaptor.forClass(AnalyzeCommentRequest.class);
+    verify(mockAPI).analyze(requestCaptor.capture()); 
+
+    AnalyzeCommentRequest actual = requestCaptor.getValue();
```

Comment:
Suggest naming this request, so that when reading the code below it's clear what this object represents.

You could also name it actualRequest, but since there's only one request in this scope, the additional adjective is not strictly needed.

Reply:
That makes sense. I think I'll name it `request`.

Reply:
Done